api/src/services/async-storage.js

```diff
@@ -0,0 +1,8 @@
+const { AsyncLocalStorage } = require('node:async_hooks');
+const asyncLocalStorage = new AsyncLocalStorage();
+const { REQUEST_ID_HEADER } = require('../server-utils');
+
+const request = require('@medic/couch-request');
+request.initialize(asyncLocalStorage, REQUEST_ID_HEADER);
+
+module.exports = asyncLocalStorage;
```

Comment:
I think this is too leaky, which is particularly dangerous because most JS devs won't be expecting a thread local obj. Can we hide the implementation behind an API so it's more like...

```js
modules.exports = {
  getRequestId = () => {
    const localStorage = asyncLocalStorage.getStore();
    return localStorage?.clientRequest?.id;
  },
  setRequestId = (req, callback) => {
    asyncLocalStorage.run({ clientRequest: req }, () => callback());
  }
}
```

Maybe this doesn't work (I haven't tested), or maybe it ends up just as complicated?